### PR TITLE
docs(python): document polars string duration language in temporal range functions

### PR DESCRIPTION
**Pull request by MarcoGorelli**

### CODE REVIEW

File: py-polars/polars/functions/range/date_range.py [hunk 1]
```diff
@@ -126,42 +122,38 @@ def date_range(
     Notes
     -----
     1) If both ``start`` and ``end`` are passed as date types (not datetime), and the
-    interval granularity is no finer than 1d, the returned range is also of
-    type date. All other permutations return a datetime Series.
+       interval granularity is no finer than 1d, the returned range is also of
+       type date. All other permutations return a datetime Series.
 
-    2) Because different months of the year have differing numbers of days, the offset
```
Comment (MarcoGorelli) [inline py-polars/polars/functions/range/date_range.py:132]: the long explanation starting here is good, but to be honest I think it's way too long to be included in a docstring

I've drastically shortened it - there's already an example of how to get the last day of each month

File: py-polars/polars/functions/range/date_range.py [hunk 2]
```diff
@@ -126,42 +122,38 @@ def date_range(
...
-    and offsets by 1 month, does the user intend to target March 28th (the next month
-    but same day), or March 31st (the end of the month)?
+       .. deprecated:: 0.19.3
 
-    Polars uses the first approach: February 28th offset by 1 month is March 28th. When
-    a date-series is generated, each date is offset as of the prior date, meaning that
-    if one began January 31st, 2023, and offset by ``1mo_saturating`` until May 31st,
-    the following dates would be generated:
+           In a future version of Polars, `date_range` will always return a `Date`.
+           Please use :func:`datetime_range` if you want a `Datetime` instead.
```
Comment (MarcoGorelli) [inline py-polars/polars/functions/range/date_range.py:131]: can't get the link to show up correctly in sphinx 🤔

Comment (stinodego) [reply py-polars/polars/functions/range/date_range.py:131]: Hm, try:
```
:func:`~polars.datetime_range`
```

Comment (MarcoGorelli) [reply py-polars/polars/functions/range/date_range.py:131]: got it, `datetime_range*` were just missing from `functions.rst` (missed that when reviewing, apologies)

**Review by stinodego — APPROVED** [review]

File: py-polars/polars/functions/range/date_range.py (hunk 2 above)
Comment (stinodego) [reply py-polars/polars/functions/range/date_range.py:131]: Ah, thanks. Good catch!

**Pull request merged**